components/gitpod-protocol/src/util/gitpod-host-url.ts

```diff
@@ -16,10 +16,10 @@ const baseWorkspaceIDRegex =
     "(([a-f][0-9a-f]{7}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})|([0-9a-z]{2,16}-[0-9a-z]{2,16}-[0-9a-z]{8,11}))";
 
 // this pattern matches v4 UUIDs as well as the new generated workspace ids (e.g. pink-panda-ns35kd21)
-const workspaceIDRegex = RegExp(`^${baseWorkspaceIDRegex}$`);
+const workspaceIDRegex = RegExp(`^(?:debug-)?${baseWorkspaceIDRegex}$`);
```

Comment:
@geropl Could someone have a look please? 🙏 

it is rather only this change for webapp. It does not really affect anythin besides supervisor frontend as far as I aware.

Debug workspaces are started within a regular workspace, not via Gitpod itself.

Reply:
> Debug workspaces are started within a regular workspace, not via Gitpod itself.

That would be an excellent doc comment on the `debugWorkspace` property below.